src/test/java/com/atlassian/db/replica/api/TestDualConnection.java

```diff
@@ -1334,4 +1337,26 @@ public void shouldClosedConnectionBeNotValid() throws SQLException {
 
         assertThat(dualConnection.isValid(1)).isFalse();
     }
+
+    @Test
+    public void shouldPropagateLastWrite() throws SQLException {
+        final ConnectionProviderMock connectionProvider = new ConnectionProviderMock();
+        final MonotonicMemoryCache<Instant> lastWriteCache = new MonotonicMemoryCache<>();
+
+        final Connection connection = DualConnection.builder(
+            connectionProvider,
+            new PessimisticPropagationConsistency
+                .Builder()
+                .assumeMaxPropagation(Duration.ofDays(10))
+                .cacheLastWrite(lastWriteCache)
+                .build()
+        ).dirtyConnectionCloseHook(Connection::commit)
+            .build();
+        connection.setAutoCommit(false);
+        connection.prepareStatement(SIMPLE_QUERY).executeUpdate();
+//        connection.commit();
```

Comment:
Do you want to leave it for the sake of visibility? Maybe some comment should be added cause sb might just thought it's a rubbish?

Reply:
Removed